content/05.dl-complexities.md

```diff
@@ -1 +1,25 @@
 ## Tip 3: Understand the complexities of training deep neural networks {#complexities}
+
+Correctly training deep neural networks is a non-trivial process.
+There are many different options and potential pitfalls at every stage.
+To get good results you have to expect to train many networks with
+different tweaks and modifications.
+Despite improved framework ease-of-use and on-demand cloud computing resources
+this means DL can be very demanding, requiring significant
+infrastructure and patitence to complete.
+The experimentation inherent to DL is often noisy (requiring repetition) and
+represent a significant organisational challenge.
+All code, parameters, and results must be carefully corralled using good
+coding practices (e.g. version control, continuous integration etc.) in order to 
+be effective and interpretable.
+This organisation is also key to being able to efficiently share your work
+and to update your model as new data becomes available. 
+
+
+Similar to [Tip 4](#baselines), try to start with a relatively smaller network
+and increase the size and complexity as needed to prevent wasting time and 
+resources. 
+Beware of the seemingly trivial choices that are being made implicitly by 
+default settings in your framework of choice e.g. choice of optimization algorithm
+(adaptive methods are often faster but can lead to worse results [@arxiv:1705.08292]).
```

Comment:
> "adaptive methods are often faster but can lead to worse results"

I'd suggest

"adaptive methods often lead to faster convergence during training but may lead to worse generalization performance on independent datasets" 

@arxiv:1705.08292: This was a 2017 NIPS paper, maybe it would be worth citing the peer-reviewed version:

Wilson, A. C., Roelofs, R., Stern, M., Srebro, N., & Recht, B. (2017). The marginal value of adaptive gradient methods in machine learning. In Advances in Neural Information Processing Systems (pp. 4148-4158).

A related one is https://arxiv.org/abs/1712.07628 by Keskar & Socher, but not sure if this has been published somewhere

Reply:
Think I managed to add the json format properly... guess I'll find out in a second once travis finishes...

Reply:
Updated to Wilson citation from the NeurIPS proceedings

Reply:
For NeurIPS proceedings, URL citations are an option so that you don't have to manually provide the JSON: `url:https://papers.nips.cc/paper/7003-the-marginal-value-of-adaptive-gradient-methods-in-machine-learning`

It doesn't work well with every conference proceedings site, but this paper looks accurate.